addons/account_fleet/models/fleet_vehicle.py

```diff
@@ -16,8 +16,20 @@ def _compute_move_ids(self):
             self.bill_count = 0
             return
 
+        query = self.env['account.move.line']._where_calc([('vehicle_id', 'in', self.ids), ('move_id.state', '!=', 'cancel')])
+        tables, where_clause, where_params = query.get_sql()
+        self.env.cr.execute('''
+            SELECT
+                vehicle_id,
+                array_agg(DISTINCT move_id)
+            FROM ''' + tables + '''
+            WHERE ''' + where_clause + '''
+            GROUP BY vehicle_id
+        ''', where_params)
+
+        vehicle_move_mapping = {vehicle_id: moves for vehicle_id, moves in self._cr.fetchall()}
         for vehicle in self:
-            vehicle.account_move_ids = self.env['account.move.line'].search([('vehicle_id', '=', vehicle.id), ('move_id.state', '!=', 'cancel')]).move_id
+            vehicle.account_move_ids = [Command.link(vehicle_move_mapping.get(vehicle.id, []))]
```

Comment:
```suggestion
            vehicle.account_move_ids = [Command.set(vehicle_move_mapping.get(vehicle.id, []))]
```
?

Reply:
I agree with the first part, it makes more sense.

I don't quite understand the second one though. Instead of what? 
This won't give the move_ids so it can't replace the SQL part right?
Is it to set the bill_count? If so, what is wrong with the current system (Just curious)

Reply:
I went too fast, I implemented array_agg_distinct in another branch a while ago I think and got confused

Doesn't `fields=['move_id:array_agg'],` work?

Reply:
array_agg_distinct would be perfect for this case :p

It does not, because if you have two lines on the same move_id with the same vehicle the id will appear twice in the resulting move_ids. (It could be removed after getting it of course, but I don't think it's perfect.)
I just re-tested this.

Reply:
Just use a `set()` on the result with `fields=['move_id:array_agg']`

Reply:
Ok I did it